pkg/backend/httpstate/backend.go

```diff
@@ -1089,6 +1089,38 @@ func (b *cloudBackend) apply(
 			colors.SpecHeadline+"%s (%s)"+colors.Reset+"\n\n"), actionLabel, stack.Ref())
 	}
 
+	// Grab any `pulumi:tag` config values and use those to update the stack's tags.
```

Comment:
This should be part of GetMergedStackTags rather than explictly getting and setting tags.